common/lib/client/auth.js

```diff
@@ -40,23 +40,31 @@ var Auth = (function() {
 
 	function Auth(client, options) {
 		this.client = client;
-		this.tokenParams = options.defaultTokenParams || {};
+		this._setOptions(options);
+	}
+
+	Auth.prototype._setOptions = function(options) {
+		this.tokenParams = this.tokenParams || options.defaultTokenParams || {};
 
-		/* RSA7a4: if options.clientId is provided and is not
-		 * null, it overrides defaultTokenParams.clientId */
 		if(options.clientId) {
-			this.tokenParams.clientId = options.clientId;
-			this.clientId = options.clientId
+			var err = this.setClientId(options.clientId);
+			if(err) { throw err; }
 		}
 
 		/* decide default auth method */
 		var key = options.key;
 		if(key) {
-			if(!options.clientId && !options.useTokenAuth) {
+			if(!this.clientId && !options.useTokenAuth) {
 				/* we have the key and do not need to authenticate the client,
 				 * so default to using basic auth */
 				Logger.logAction(Logger.LOG_MINOR, 'Auth()', 'anonymous, using basic auth');
+				if(this.method && this.method !== 'basic') {
```

Comment:
For clarity, should we not be writing these conditionals as `if(this.method && (this.method !== 'basic') {`?